### DISCUSSION TITLE
Adding additional Maven repositories in Gradle projects

### DISCUSSION THREAD

**Discussion by dogboy21** [renovatebot/renovate#18930] [Request Help]
Hi,

I'm currently trying to setup Renovate in our environment with Gradle projects.
We're using a custom Gradle plugin to add some custom common settings and repositories into our Gradle projects.
Since Renovate only parses the Gradle files of the project, it does not pick up the additional repositories of the plugin for dependency lookups.

Although [the documentation](https://docs.renovatebot.com/java/#custom-registry-support-and-authentication) states that it should be possible to add additional repositories, I haven't found any solution yet.
Adding the repository to the `registryUrls` in the `packageRules` completely overrides all scanned repositories of the Gradle project.

Is there any solution to keep the repositories of the Gradle project while still adding additional repositories?

Thanks!

**Comment by viceice** [comment]
not possible, you need to override all via renovate config.

**Reply by clrxbl** [reply]
Would it be possible to get this feature implemented at some point? Overriding all really isn't a great solution and not everyone has a Maven repository that mirrors/proxies public Maven repositories (which realistically would be the only way to use this feature in this case).

**Reply by rarkins** [reply]
It's not clear to me what the feature even would be. Renovate cannot be expected to understand any custom plugin which a user has.

**Reply by dogboy21** [reply]
Which is totally fine and not even needed in my opinion.
A simple option similar to `registryUrls` to define repository urls in addition to the dynamically discovered repositories instead of completely overriding them should be enough

**Reply by rarkins** [reply]
Internally there is already an option called `additionalRegistryUrls` which functions according to what you're describing. Are you saying you'd need the ability to manually specify these in config?

**Reply by dogboy21** [reply]
Yes, that would be great!

**Reply by rarkins** [reply]
Created #18979
